### DISCUSSION TITLE
How to get "Then" to fire for each Order (Getting Started Sample)

### DISCUSSION THREAD

**Discussion by tcoolspy** [NRules/NRules#268] [General]
Looking at the getting started sample, I notice there are 2 orders for the customer.  The "NotifyAboutDiscount" method is fired as expected, but what if I wanted to send a notice for each order?  In other words, how do we get the "Do" method to fire for each item in a collection?  I hope this makes sense.

Thanks.

**Comment by tcoolspy** [comment]
After I asked this question, I realized that if I modified the example to create a new session for each order (with the customer in the session of course) the notification fires for each order in that scenario, but I was wondering if the notification could fire for each order within a given session.

**Comment by snikolayev** [comment]
@tcoolspy if you wanted the rule to fire for each order, use Match instead of Exists for orders

**Reply by tcoolspy** [reply]
Awesome, thanks.
